WheelOfFortune/Prompt.cs

```diff
@@ -121,25 +123,45 @@ public static dynamic PromptAction(Action action)
         private static string validateUserGuess(string userInput)
         {
             userInput = userInput.ToLower();
-            if (userInput.Length != 1)
-            {
-                Console.WriteLine("Invalid Guess.Please enter a Non Empty,Single letter:");
-                userInput = Console.ReadLine();
-                return userInput;
-            }
-            else
+            InvalidInputError error = null;
+            bool valid = false;
+            do
             {
-                return userInput;
+                if(userInput.Length != 1)
+                {
+                    if (userInput.Length == 1)//checking if the input is empty
+                    {
+                        error = new InvalidInputError(InvalidInputError.ErrorTypes.EmptyInput);
+
+                    }
+                    else if (userInput.Length > 1)//checking if the input is Multi Letters
+                    {
+                        error = new InvalidInputError(InvalidInputError.ErrorTypes.ReceivedStringExpectedChar);
+                    }
+                    Console.Write($"\n{error.ErrorMessage}");
+                    valid = false;
+                    Console.Write("Enter a letter:");
+                    userInput = Console.ReadLine();
+                }
+                else
+                {
+                    valid = true;
+                }
             }
+            while (!valid);
+            
+            return userInput;
         }
 
 
         /// <summary>
-        /// A method to prompt to the user that the input is invalid
+        /// A method to prompt to the user that the input is invalid if an empty string is entered as a puzzle solution
         /// </summary>
-        public static string PromptInValidEmptyInput()
+        public static string PromptInValidEmptyInputAsPuzzleSolution()
         {
-            Console.WriteLine("The Input can't be empty.Please enter a valid solution to the puzzle: ");
+            InvalidInputError error = new InvalidInputError(InvalidInputError.ErrorTypes.EmptyInput);
+            Console.Write($"\n{error.ErrorMessage}");
```

Comment:
This could use WriteLine so that the user will first see the error message, then on a new line, be prompted to give a valid solution